algonaut_transaction/src/api_model.rs

```diff
@@ -614,4 +733,57 @@ mod tests {
 
         assert_eq!(lsig, lsig_deserialized);
     }
+
+    #[test]
+    fn test_api_box_references_from_box_references() {
+        let box_name = vec![1, 2, 3, 4];
+        let current_app_id = 6355;
+        let foreign_apps = vec![8577, 7466];
+        let boxes = vec![
+            BoxReference {
+                app_id: Some(6355),
+                name: box_name.clone(),
+            },
+            BoxReference {
+                app_id: Some(7466),
+                name: box_name.clone(),
+            },
+            BoxReference {
+                app_id: Some(8577),
+                name: box_name.clone(),
+            },
+        ];
+
+        let api_boxes = api_box_references_from_box_references(
+            Some(&boxes),
+            Some(&foreign_apps),
+            Some(current_app_id),
+        )
+        .expect("api box references can be created");
```

Comment:
nit: Rust's recommended message style for `expect` would be something like "api box should be created" (ref: https://doc.rust-lang.org/std/option/enum.Option.html#recommended-message-style)